Yoast/javascript
# Have less buttons and manage more via props

## Summary


This PR can be summarized in the following changelog entry:

* Yoast styled buttons are now available in preset variants, such as `primary`, `secondary`, `buy`, etc.
* Yoast styled links (identically styled to the buttons) are now available in preset variants, such as `primary`, `secondary`, `buy`, etc.

## Relevant technical choices:

* RTL styling is off, not fixed in this PR.
* `SvgIcon` was not suited to work for us here, especially given the workflow `team-frontend` desires.
* Adding icons to buttons works by setting them in CSS as an image mask on distinct class. Then, you add a variant name to the `variantToIcon` object near the top of `Button.js` like so: 

```js
variantName: { iconAfter: "yoast-button--classNameOfSpanWithMask" }
```
(use `iconBefore` if you need the span with the icon to appear in front of the text ).

* Add variants is as simple as adding the variant name as a key to `iconToClassName` and setting the desired class as a value. The often occurring `yoast-button yoast-button--` is available as a constant.

## Test instructions

This PR can be tested by following these steps:

* Checkout the components app, and verify that all button and links variants look and work as they should.
* Perhaps try to add some variants yourself, to see if it "feels" nice to work with them.
* Note that RTL is (I think) erroneously implemented, so that needs some work once we get to these buttons in the future.

## Quality assurance


Fixes #
